src/Builder/Visitor/AddMethodCallToCallListVisitor.php

```diff
@@ -0,0 +1,197 @@
+    {
+        $calledMethods = $this->getCalledMethods($node);
+        $callMetadataTransfer = $this->classMetadataTransfer->getCall();
+        if (!$callMetadataTransfer) {
+            return $node;
+        }
+
+        $before = $callMetadataTransfer->getBefore();
+        if ($before) {
+            $beforeMethodName = $this->getMethodNameFromNamespaceName($before);
+            $beforePosition = array_search($beforeMethodName, $calledMethods, true);
+            if (is_int($beforePosition)) {
+                return $this->addNewCallIntoArrayMergeFuncNodeByPosition($node, $beforePosition);
+            }
+        }
+
+        $after = $callMetadataTransfer->getAfter();
+        if ($after) {
+            $afterMethodName = $this->getMethodNameFromNamespaceName($after);
+            $afterPosition = array_search($afterMethodName, $calledMethods, true);
+            if (is_int($afterPosition)) {
+                return $this->addNewCallIntoArrayMergeFuncNodeByPosition($node, $afterPosition + 1);
+            }
+        }
+
+        $node->args[] = $this->createFunctionArgWithMethodCall();
+
+        return $node;
+    }
+
+    /**
+     * @param \PhpParser\Node\Expr\FuncCall $node
+     * @param int $position
+     *
+     * @return \PhpParser\Node
+     */
+    protected function addNewCallIntoArrayMergeFuncNodeByPosition(FuncCall $node, int $position): Node
+    {
+        array_splice($node->args, $position, 0, [$this->createFunctionArgWithMethodCall()]);
+
+        return $node;
+    }
+
+    /**
+     * @param \PhpParser\Node\Expr\FuncCall $node
+     *
+     * @return array<string>
+     */
+    protected function getCalledMethods(FuncCall $node): array
+    {
+        $result = [];
+
+        $nodeFinder = new NodeFinder();
+        /** @var array<\PhpParser\Node\Expr\MethodCall> $calledMethods */
+        $calledMethods = $nodeFinder->findInstanceOf($node, MethodCall::class);
+        foreach ($calledMethods as $method) {
+            /** @var \PhpParser\Node\Identifier $methodName */
+            $methodName = $method->name;
+            $result[] = $methodName->name;
+        }
+
+        return $result;
+    }
+
+    /**
+     * @return \PhpParser\Node\Arg
+     */
+    protected function createFunctionArgWithMethodCall(): Arg
+    {
+        $var = new Variable(static::THIS_KEY);
+        $methodName = $this->classMetadataTransfer->getTargetMethodNameOrFail();
+        $methodCall = new MethodCall($var, $methodName);
+
+        return new Arg($methodCall);
+    }
+
+    /**
+     * @return int
+     */
+    protected function successfullyProcessed(): int
+    {
+        $this->methodFound = false;
+
+        return NodeTraverser::DONT_TRAVERSE_CHILDREN;
+    }
+
+    /**
+     * @param string $namespaceName
+     *
+     * @return string
+     */
+    protected function getMethodNameFromNamespaceName(string $namespaceName): string
+    {
+        $data = explode('::', $namespaceName);
```

Comment:
```
return end(explode('::', $namespaceName));
```

Reply:
Error: only variables can be passed to `end` functions